### PR TITLE
fix: add public inpuput to allow provenance

### PR DESCRIPTION
**Pull request by claudia-codes**
Solves #301 

To un-draft this: 
- [ ] solution must be validated
- [ ] tests must pass

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -284,7 +291,9 @@ jobs:
...
-| `provenance`| No    | Set as true to have NPM [generate a provenance statement](https://docs.npmjs.com/generating-provenance-statements). See [Provenance section above](#provenance) for requirements.<br /> (_Default: `false`_)                                                                  |
+| `provenance`| No    | Set as true to have NPM [generate a provenance statement](https://docs.npmjs.com/generating-provenance-statements). See [Provenance section above](#provenance) for requirements.<br /> (_Default: `false`_)      
+                                                                                                                   |
+| `provenance`| No    | Set as `public` to allow [generate a provenance statement](https://docs.npmjs.com/generating-provenance-statements). See [Provenance section above](#provenance) for requirements.<br /> (_Default: `private`_)                                                                  |
```
Comment (AlanSl) [inline README.md:296]:
```suggestion
| `access`| No    | Set as `public` to allow [generate a provenance statement](https://docs.npmjs.com/generating-provenance-statements). See [Provenance section above](#provenance) for requirements.<br /> (_Default: `private`_)                                                                  |
```

**Review by simoneb — COMMENTED** [review]
so far the implementation looks good if we are in agreement that we can't just enable public access implicitly, which I'm still not 100% sure about and I'd have to check. in any case, so far so good

File: src/utils/publishToNpm.js [hunk 2]
```diff
@@ -41,24 +41,47 @@ async function allowNpmPublish(version) {
...
     'set',
     `//registry.npmjs.org/:_authToken=${npmToken}`,
   ])
 
+  let packageName = null
+  try {
+    const packageInfo = await execWithOutput('npm', ['view', '--json'])
+    packageName = packageInfo ? JSON.parse(packageInfo).name : null
+  } catch (error) {
+    if (!error?.message?.match(/code E404/)) {
```
Comment (AlanSl) [inline src/utils/publishToNpm.js:76]: Nit, but I wonder if we can avoid duplicating this, e.g. have the function that calls it already return a status number. or pull it out into its own utility that passes the "is published" state into `allowNpmPublish`?

**Review by AlanSl — COMMENTED** [review]
I think the approach makes sense, though one question in addition to the other comments: are we expecting to add a feature where changing the `access` setting actually changes the access on NPM?

It feels like having this option creates that expectation, so if that's not part of the plan it needs to be clearly documented I think.

File: README.md (hunk 1 above)
Comment (AlanSl) [reply README.md:296]: That's what I meant, "_Default: `private`_" seems to imply that if it is not defined, it'll make a package private.

**Comment by simoneb** [comment]
I'm not sure we should be going ahead with this change because the requirements pertaining provenance and non-public packages are not clear.

If we want to create a new "access" input, then we should implement it fully, meaning that it should always be provided to the npm publish command, regardless of whether we use provenance or not.

I think we need to experiment more with this. E.g. does having a "private" property in package.json affect this behavior and would it be enough to publish with provenance without having to change this action? I don't know.

**Pull request closed without merging**